src/MICore/Transports/LocalUnixTerminalTransport.cs

```diff
@@ -121,6 +124,8 @@ public override void Close()
             // If we are shutting down before the _exitReader has read a line it's possible that
             // there is a thread blocked doing a read() syscall. 
             ForceDisposeStreamReader(_exitReader);
+            _streamReadPidCancellationTokenSource.Cancel();
+            _streamReadPidCancellationTokenSource.Dispose();
```

Comment:
In AD7, this isn't a scenario that exists (launch is sync). I think this would be my suggestion:

1. AD7Engine - make this implement a new IVSCodeEngineLaunch interface which we can define in the host assembly. This isn't the best place for it, but it seems like the best option to me. Ideally this would have a very different signature than the current LaunchSuspended method, as I think the current LaunchSuspended method is actually the wrong thing for VS Code. But for now we can just fix the obvious problem with it and add a CancellationToken to it but otherwise leave it unchanged.
2. Plumb on through the cancellation token
3. In OpenDebugAD7, add in a copy of the IVSCodeEngineLaunch cs file
4. Set the launch cancellation token if stop debugging is hit